sonar-csharp-plugin/src/test/java/org/sonar/plugins/csharp/CSharpFileCacheSensorTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * SonarC#
+ * Copyright (C) 2014-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.csharp;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.fs.internal.TestInputFileBuilder;
+import org.sonar.api.batch.sensor.cache.WriteCache;
+import org.sonar.api.batch.sensor.internal.DefaultSensorDescriptor;
+import org.sonar.api.batch.sensor.internal.SensorContextTester;
+import org.sonar.api.config.internal.MapSettings;
+import org.sonar.api.utils.log.LogTester;
+import org.sonar.api.utils.log.LoggerLevel;
+import org.sonarsource.dotnet.shared.plugins.HashProvider;
+
+import java.io.File;
+import java.io.IOException;
+import java.security.NoSuchAlgorithmException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class CSharpFileCacheSensorTest {
+  @Rule
+  public TemporaryFolder temp = new TemporaryFolder();
+
+  @Rule
+  public LogTester logTester = new LogTester();
+
+  @Test
+  public void should_describe() {
+    var sensorDescriptor = new DefaultSensorDescriptor();
+    var sensor = new CSharpFileCacheSensor(null, new HashProvider());
+    sensor.describe(sensorDescriptor);
+
+    assertThat(sensorDescriptor.name()).isEqualTo("C# file caching sensor");
```

Comment:
Other sensors use this casing
```suggestion
    assertThat(sensorDescriptor.name()).isEqualTo("C# File Caching Sensor");
```

Reply:
I'm not sure if it should be C# or CSharp, let's use `pluginMetadata.shortLanguageName()` or `pluginMetadata.languageName` - we're inconsistent in their usage :/

Reply:
hmm, we shouldn't. I've added a note to address this.

Reply:
for the context: `pluginMetadata.shortLanguageName()` is a redirect to `CSharpPlugin/VbNetPlugin.LANGUAGE_NAME` which is the same with `CSharp.getName()/VbNet.getName()`. A single constant exposed by all this.